ipn/ipnlocal/local.go

```diff
@@ -140,9 +141,23 @@ type LocalBackend struct {
 	statusChanged *sync.Cond
 }
 
+type clientGen func(controlclient.Options) (controlclient.Client, error)
+
 // NewLocalBackend returns a new LocalBackend that is ready to run,
 // but is not actually running.
 func NewLocalBackend(logf logger.Logf, logid string, store ipn.StateStore, e wgengine.Engine) (*LocalBackend, error) {
+	// TODO(apenwarr): change controlclient.New to return controlclient.Client?
```

Comment:
This whole todo goes away if callers provide the controlclient they want IPN to use :) Then the controlclient library can return a concrete type, ipn can take in an interface, and everyone's happy.

Reply:
As mentioned elsewhere, ipn.Backend needs to be able to shut down and re-create the controlclient on demand when switching sessions. This is the minimal change needed to change what it constructs, so it doesn't really make things worse. We can fix it up later.